From eeee78f39f8206ad2ff5cfbc0715590a2cdd403c Mon Sep 17 00:00:00 2001
From: Arun Ravindran <arun.ravindran@intel.com>
Date: Tue, 30 Apr 2013 16:09:33 +0300
Subject: [PORT FROM MAIN] Telephony: Clear CallForwarding and Message waiting Indications

BZ: 64610

When SIM is removed the SIMRecords object is disposed, but
the relevant events to clear the Callforwarding and message
waiting icons are not send.

This patch clears the call forwarding and message waiting
indicators by clearing the internal member data and sending
out notifications for the change in data.

Change-Id: I9a123c8fbeccf2c6a11d1d60a58eeb9898f25cdb
Orig-Change-Id: I05e969ed197c0eedfc87f83189873aa370b48005
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../internal/telephony/uicc/SIMRecords.java        |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/uicc/SIMRecords.java b/src/java/com/android/internal/telephony/uicc/SIMRecords.java
index 6b6c7f2..c77a107 100644
--- a/src/java/com/android/internal/telephony/uicc/SIMRecords.java
+++ b/src/java/com/android/internal/telephony/uicc/SIMRecords.java
@@ -225,6 +225,10 @@ public class SIMRecords extends IccRecords {
         mSpn = null;
         mEfMWIS = null;
         mEfCPHS_MWI = null;
+        setVoiceMessageWaiting(1, 0);
+        mEfCfis = null;
+        mEfCff = null;
+        setVoiceCallForwardingFlag(1, false);
         mSpdiNetworks = null;
         mPnnHomeName = null;
         mGid1 = null;
-- 
1.7.4.1

